### PR TITLE
Fixes and Rev4 updates for filesystem controls

### PR DESCRIPTION
**Pull request by samcornwell**
These are all the changes we have made in our repo at https://github.com/djhaynes/inspec-profile-disa_stig-el7 relating to file system controls. This includes updates of these controls to RHEL7 STIG Rev4, as well as various fixes.

### CODE REVIEW

File: controls/V-72007.rb [hunk 1]
```diff
@@ -1,54 +1,45 @@
...
+directories on the system with the \"chown\" command:
 
 # chown <user> <file>"
+  tag "fix_id": "F-78359r1_fix"
+
+  findings = Set[]
+  partitions = etc_fstab.params.map{|partition| partition['file_system_type']}.uniq
+
+  partitions.each do |part|
+    findings = findings + command("find / -xdev -fstype #{part} -nouser").stdout.lines.map(&:chomp)
```
Comment (trevor-vaughan) [inline controls/V-72007.rb]: This should also have `-xautofs`

File: controls/V-72009.rb [hunk 2]
```diff
@@ -1,55 +1,40 @@
...
+  tag "fix": "Either remove all files and directories from the system that do
+not have a valid group, or assign a valid group to all files and directories on
+the system with the \"chgrp\" command:
 
 # chgrp <group> <file>"
+  tag "fix_id": "F-78361r1_fix"
 
-  describe command('find / -xdev -fstype xfs -nogroup') do
-    its('stdout.strip') { should match /^$/ }
+  describe command('find / -fstype xfs -nogroup') do
```
Comment (trevor-vaughan) [inline controls/V-72009.rb]: This needs to loop through all valid filesystems on the system that are not `tmpfs` or `sysfs` (so far).

File: controls/V-72059.rb [hunk 3]
```diff
@@ -1,54 +1,47 @@
...
+)
 
-Release Date: 2017-03-08
-Version: 1
-Publisher: DISA
-Source: STIG.DOD.MIL
-uri: http://iase.disa.mil
------------------
-=end
+NON_INTERACTIVE_SHELLS = attribute(
```
Comment (trevor-vaughan) [inline controls/V-72059.rb]: No globals

File: controls/V-72061.rb [hunk 4]
```diff
@@ -1,52 +1,35 @@
...
 
 # grep /var /etc/fstab
-UUID=c274f65f    /var                    ext4    noatime,nobarrier        1 2
+UUID=c274f65f /var ext4 noatime,nobarrier 1 2
 
 If a separate entry for \"/var\" is not in use, this is a finding."
   tag "fix": "Migrate the \"/var\" path onto a separate file system."
+  tag "fix_id": "F-78413r1_fix"
 
   # @todo fstab resource?
```
Comment (trevor-vaughan) [inline controls/V-72061.rb]: The `fstab` entry is largely irrelevant to the reality of the system configuration.

**Comment by samcornwell** [comment]
@trevor-vaughan I've addressed your concerns, should be good to go.

File: controls/V-72059.rb [hunk 5]
```diff
@@ -58,11 +58,11 @@
 system/partition."
   tag "fix_id": "F-78411r1_fix"
 
-  IGNORE_SHELLS = NON_INTERACTIVE_SHELLS.join('|')
+  ignore_shells = non_interactive_shells.join('|')
 
   # excluding root because its home directory is usually "/root" (mountpoint "/")
-  users.where{ !shell.match(IGNORE_SHELLS) && (uid >= 1000)}.entries.each do |user_info|
-    next if EXEMPT_HOME_USERS.include?("#{user_info.username}")
+  users.where{ !shell.match(ignore_shells) && (uid >= 1000)}.entries.each do |user_info|
```
Comment (trevor-vaughan) [inline controls/V-72059.rb]: This should be `UID_MIN` from the target system instead of `1000` since that can be arbitrarily set by the user.

**Comment by samcornwell** [comment]
@trevor-vaughan alright, done

**Pull request merged**